### PR TITLE
Stringio, added exists, handle regions

### PR DESCRIPTION
**Pull request by bendangelo**

### CODE REVIEW

File: lib/bunny_storage_client.rb [hunk 1]
```diff
@@ -49,6 +52,11 @@ def get_file(storage_zone_name: nil, filename: nil, as: :string)
     request['accept'] = '*/*'
     response = make_request(uri, request)
 
+    if response.code != '200'
+      @logger.error("Failed to get file from #{storage_zone_name}/#{filename}: #{response.code} #{response.body}")
+      return nil
+    end
```
Comment (rkwap) [inline lib/bunny_storage_client.rb]:
```suggestion
    raise StandardError, "#{response.code} #{response.body}" unless success_code?(response.code)
```

Keeping consistent with a single logger statement.

**Review by rkwap — CHANGES_REQUESTED** [review]
Small refactor. Other changes LGTM!

**Comment by rkwap** [comment]
@bendangelo Apologies for the late review. Changes LGTM. Thanks for contributing!
Do test changes and leave a comment if everything works fine!
Will merge this ASAP.

File: lib/bunny_storage_client.rb (hunk 1 above)
Comment (bendangelo) [reply lib/bunny_storage_client.rb]: Hey throwing an error will break the `exist?` and other methods. For this method a soft fail is best unless you have an other idea.

Comment (rkwap) [reply lib/bunny_storage_client.rb]: I'm not sure about your concern. Here's why:

1. The exist? method has its own independent implementation and doesn't rely on get_file at all.
2. From an execution perspective, the end result will be identical - the rescue block in this method already handles any exceptions by soft failing (returning nil). Adding a raise here won't change that behavior.

**Comment by bendangelo** [comment]
ok done

**Comment by rkwap** [comment]
Thank you for your contribution :)
merging!

**Pull request merged**